### PR TITLE
Introduce aliases version

### PR DESCRIPTION
**Pull request by jasontedor**
This commit introduces aliases versions to index metadata. This will be useful in CCR when we replicate aliases.

Relates #41396

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java [hunk 1]
```diff
@@ -684,6 +711,11 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             version = in.readLong();
             mappingVersion = in.readVLong();
             settingsVersion = in.readVLong();
+            if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
+                aliasesVersion = in.readVLong();
+            } else {
+                aliasesVersion = 1;
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java]:
```suggestion
                aliasesVersion = 0;
```

**Comment by jasontedor** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by jasontedor** [comment]
Thanks for reviewing @martijnvg.

**Pull request merged**